### PR TITLE
permissions: allow moderators to read emails

### PR DESCRIPTION
**Pull request by kpsherva**
* closes https://github.com/inveniosoftware/invenio-requests/issues/347
* closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1411

# FAQ
- administration of users was moved to app-rdm because of common components with users moderations (which requires requests) - I found it better to move it "lower" than causing circular imports

### CODE REVIEW

File: invenio_users_resources/records/mappings/os-v1/users/user-v1.0.0.json [hunk 1]
```diff
@@ -71,7 +102,12 @@
             "type": "text"
           },
           "affiliations": {
-            "type": "text"
+            "type": "text",
+            "fields": {
+              "keyword": {
```
Comment (kpsherva) [inline invenio_users_resources/records/mappings/os-v1/users/user-v1.0.0.json:107]: needed for aggs

File: invenio_users_resources/entity_resolvers.py [hunk 2]
```diff
@@ -73,7 +75,12 @@ def pick_resolved_fields(self, identity, resolved_dict):
                 "full_name": profile.get("full_name", ""),
                 "affiliations": profile.get("affiliations", ""),
             },
-            "links": {"avatar": avatar},
+            "links": resolved_dict.get("links"),
+            "blocked_at": resolved_dict.get("blocked_at"),
+            "verified_at": resolved_dict.get("verified_at"),
```
Comment (kpsherva) [inline invenio_users_resources/entity_resolvers.py:80]: to be confirmed if we can expand those fields

Comment (slint) [reply invenio_users_resources/entity_resolvers.py:80]: My hunch is that if we're using `email` as above,  this goes through the service and gets properly filtered, in which case any attribute would be good.

Comment (kpsherva) [reply invenio_users_resources/entity_resolvers.py:80]: true! thanks

**Pull request merged**